graphs/iterative_deepening.py

```diff
@@ -0,0 +1,78 @@
+"""
+Finds paths in the following example map to illustrate Iterative Deepening Search:
+http://www.cse.unsw.edu.au/~cs3411/16s1/tut/wk03romania.gif
+"""
+
+
+def ids(graph, start, goal):
+    # dfs helper function to perform depth limited dfs
+    def dfs(graph, start, goal, depth, path):
+        """
+        if depth < 0 , it means node is not found with current depth given return -1
+        to indicate not found
+        """
+        if depth < 0:
+            return -1
+        path.append(start)
+        if start == goal:
+            return path
+        for i in graph[start]:
+            tmp = dfs(graph, i[1], goal, depth - 1, list(path))
+            if tmp != -1:
+                return tmp
+        """
+        if any of the above dfs runs do not give path,
+        then it means node does not exist and return -1
+        """
```

Comment:
Don't mix docstring with comments.
Use `""" ... """` for docstrings and `# ... ` for comments